tests/NgTests/RegistrationCollectorTests.cs

```diff
@@ -154,5 +158,109 @@ public static IEnumerable<object[]> PageContent
                 };
             }
         }
+
+        [Fact]
+        public async Task CreatesRegistrationsWithSemVer2()
+        {
+            // Arrange
+            var catalogStorage = Catalogs.CreateTestCatalogWithSemVer2Package();
```

Comment:
nitpick:

a lot of the code here is nearly identical to that in `SharedInit`, just with the added SemVer2 storage. I would suggest modifying `SharedInit` to support SemVer2 storage as well (`private void SharedInit(bool useSemVer2)`.